xbmc/Application.cpp

```diff
@@ -2657,7 +2678,21 @@ void CApplication::FrameMove(bool processEvents, bool processGUI)
       m_pInertialScrollingHandler->ProcessInertialScroll(frameTime);
       CSeekHandler::GetInstance().Process();
     }
+
+    // Open the door for external calls e.g python exactly here.
+    // We try to close the door latest after 10 ms.
+    unsigned int maxLoops = 5;
+    m_ProcessedExternalCalls = 0;
+    while ((m_WaitingExternalCalls || m_ProcessedExternalCalls) && maxLoops--)
+    {
+      m_ProcessedExternalCalls = 0;
+      m_frameMoveGuard.unlock();
+      Sleep(2);
```

Comment:
this window should start at zero. assume there is only a single python call waiting, that only takes some micro secs to be completed. we can expand the window if there are threads waiting for x times in a row when calling FrameMove.

This code blocks the main thread at least for 4ms  :(